stew/ranges/typedranges.nim

```diff
@@ -24,16 +24,22 @@ type
   MutByteRange* = MutRange[byte]
 
 proc isLiteral[T](s: seq[T]): bool {.inline.} =
-  type
-    SeqHeader = object
-      length, reserved: int
-  (cast[ptr SeqHeader](s).reserved and (1 shl (sizeof(int) * 8 - 2))) != 0
+  when defined(gcOrc) or defined(gcArc):
+    false
+  else:
+    type
+      SeqHeader = object
+        length, reserved: int
+    (cast[ptr SeqHeader](s).reserved and (1 shl (sizeof(int) * 8 - 2))) != 0
 
 proc toImmutableRange[T](a: seq[T]): Range[T] =
   if a.len != 0:
     when rangesGCHoldEnabled:
-      if not isLiteral(a):
-        shallowCopy(result.gcHold, a)
+      when declared(shallowCopy):
+        if not isLiteral(a):
+          shallowCopy(result.gcHold, a)
+        else:
+          result.gcHold = a
```

Comment:
this defeats the purpose of the ranges entirely - I'd probably go for deprecating the whole library since there's some work on view types being done in the language, but @zah might have some plans?

Reply:
Can we merge this & move this into an issue? Or I can revert the changes to typedranges only fix results, but would be nice to be able to use results with devel